src/pedersen.rs

```diff
@@ -221,172 +58,87 @@ pub fn pedersen_hash(a: Fp, b: Fp) -> Fp {
     result = result.add(&tmp);
 
     // Return x-coordinate
-    result.x
+    StarkHash::from(result.x)
+}
+
+impl From<StarkHash> for FieldElement {
+    fn from(hash: StarkHash) -> Self {
+        debug_assert_eq!(
+            std::mem::size_of::<FieldElement>(),
+            std::mem::size_of::<StarkHash>()
+        );
+        Self::from_repr(FieldElementRepr(hash.to_be_bytes())).unwrap()
+    }
 }
 
+impl From<FieldElement> for StarkHash {
+    fn from(fp: FieldElement) -> Self {
+        debug_assert_eq!(
+            std::mem::size_of::<FieldElement>(),
+            std::mem::size_of::<StarkHash>()
+        );
+        // unwrap is safe because the FieldElement and StarkHash
+        // should both be 251 bits only.
+        StarkHash::from_be_bytes(fp.to_repr().0).unwrap()
+    }
+}
 #[cfg(test)]
 mod tests {
     use super::*;
     use pretty_assertions::assert_eq;
 
-    mod to_le_bits_rev {
-        use super::*;
-        use pretty_assertions::assert_eq;
-
-        #[test]
-        fn zero() {
-            let zero = Fp::zero().into_bits();
-            let expected = BitArray::<Lsb0, [u64; 4]>::default();
-
-            assert_eq!(zero, expected);
-        }
-
-        #[test]
-        fn one() {
-            let one = Fp::one().into_bits();
-
-            let mut expected = BitArray::<Lsb0, [u64; 4]>::default();
-            expected.set(0, true);
+    #[test]
+    fn hash() {
+        // Test vectors from https://github.com/starkware-libs/crypto-cpp/blob/master/src/starkware/crypto/pedersen_hash_test.cc
+        let a = "03d937c035c878245caf64531a5756109c53068da139362728feb561405371cb";
+        let b = "0208a0a10250e382e1e4bbe2880906c2791bf6275695e02fbbc6aeff9cd8b31a";
+        let expected = "030e480bed5fe53fa909cc0f8c4d99b8f9f2c016be4c41e13a4848797979c662";
 
-            assert_eq!(one, expected);
+        fn parse_hex(str: &str) -> [u8; 32] {
```

Comment:
TIL this naming is allowed! My editor will color this confusingly as it doesn't yet support rust-analyzers colors but no matter :)